### PR TITLE
[SPARK-44785][SQL][CONNECT] Convert common alreadyExistsExceptions and noSuchExceptions

### PR DESCRIPTION
**Pull request by heyihong**
### What changes were proposed in this pull request?

- Convert common alreadyExistsExceptions and noSuchExceptions
- Extend common alreadyExistsExceptions and noSuchExceptions to support (message, cause) constructors

### Why are the changes needed?

- Better compatibility with the existing control flow
- Better readability of errors

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

`build/sbt "connect-client-jvm/testOnly *ClientE2ETestSuite"`

### CODE REVIEW

File: connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/GrpcExceptionConverter.scala [hunk 1]
```diff
@@ -62,14 +63,20 @@ private[client] object GrpcExceptionConverter extends JsonUtils {
...
+      : (String, (String, Option[Throwable]) => Throwable) = {
     val className = implicitly[reflect.ClassTag[T]].runtimeClass.getName
     (className, throwableCtr)
   }
 
   private val errorFactory = Map(
     errorConstructor((message, _) => new ParseException(None, message, Origin(), Origin())),
-    errorConstructor((message, cause) => new AnalysisException(message, cause = Option(cause))))
+    errorConstructor((message, cause) => new AnalysisException(message, cause = cause)),
+    errorConstructor((message, _) => new NamespaceAlreadyExistsException(message)),
```
Comment (cloud-fan) [inline connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/GrpcExceptionConverter.scala:75]: is it to match spark sql?

Comment (heyihong) [reply connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/connect/client/GrpcExceptionConverter.scala:75]: Yes, the current idea is to convert the common exceptions for better compatibility with the existing control flow that catch exceptions from spark sql.

**Comment by cloud-fan** [comment]
LGTM if all tests pass

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master and branch-3.5.

**Pull request closed without merging**